### PR TITLE
Only include the necessary cost models during transaction finalization

### PR DESCRIPTION
**Pull request by jy14898**
Fixes being able to run scripts (well, some of the examples now) on the `babbage` branch

### Pre-review checklist

- [x] All code has been formatted using our config (`make format` for Purescript, `nixpkgs-fmt` for Nix)
- [x] All Purescript imports are explicit, including constructors
- [ ] **All** existing examples have been run locally against a fully-synced testnet node
- [x] The integration and unit tests have been run (`npm run test`) locally
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`)

### CODE REVIEW

File: src/Serialization.purs [hunk 1]
```diff
@@ -830,15 +830,15 @@ convertValue val = do
...
-      costModel <-
-        convertCostModel <=< fromJustEff
-          ("`" <> show lang <> "` not found in `Costmdls`")
-          $ Map.lookup lang cs
-      costmdlsSetCostModel costmdls lang' costModel
-  loadCostModel S.PlutusV1 =<< newPlutusV1
-  loadCostModel S.PlutusV2 =<< newPlutusV2
+  -- TODO Check the traversal order of the costmodels map isn't important for
+  --      mixed language transactions
+  --      https://github.com/Plutonomicon/cardano-transaction-lib/issues/764
```
Comment (klntsky) [inline src/Serialization.purs:835]: As long as I'm not missing something obvious, the cost models go to different keys, so why would there be any problem? I think this TODO is not needed, as well as the issue

Comment (jy14898) [reply src/Serialization.purs:835]: I'm not sure how costmodels is serialized when hashing as part of the transaction, which might make things not line up when hashes are being checked if order is important?

Comment (klntsky) [reply src/Serialization.purs:835]: OK but what is the reason you think the order may not be fixed?

Comment (jy14898) [reply src/Serialization.purs:835]: I don't have any information for or against ordering, and I'm not sure whose responsibility the order is, ours or CSL? I don't mind dropping the TODO if you'd like, I thought it didn't hurt to be more rigorous and also to include a hint if the issue does arise later

Comment (klntsky) [reply src/Serialization.purs:835]: > whose responsibility the order is, ours or CSL?

CSL's. Costmdls is a wrapper over `std::collections::BTreeMap` (rust) and it's a sorted map.

Comment (jy14898) [reply src/Serialization.purs:835]: OK great, I'll drop the todo then :+1:

**Review by klntsky — APPROVED** [review]
Great, thanks!

**Pull request merged**